packages/sdk/src/libs/clients/registry/client.ts

```diff
@@ -0,0 +1,45 @@
+import assert from "assert";
+import { Registry__factory, Registry } from "@uma/core/contract-types/ethers";
+import RegistryArtifacts from "@uma/core/build/contracts/Registry.json";
+import type { SignerOrProvider, GetEventType } from "../..";
+import { Event } from "ethers";
+
+type NewContractRegistered = GetEventType<Registry, "NewContractRegistered">;
+
+export interface EventState {
+  contracts?: { [key: string]: NewContractRegistered };
+}
+
+export type Network = keyof typeof RegistryArtifacts.networks;
+
+export function getAddress(network: Network): string {
+  const address = RegistryArtifacts?.networks?.[network]?.address;
+  assert(address, "no address found for network: " + network);
+  return address;
+}
+
+export type Instance = ReturnType<typeof Registry__factory.connect>;
+export function connect(address: string, provider: SignerOrProvider): Instance {
+  return Registry__factory.connect(address, provider);
+}
```

Comment:
Shouldn't `Instance` just be the `Registry` type that you import at the top? Can we drop the `Instance` type altogether in favor of `Registry`?

Reply:
good call!